atc/configwarning.go

```diff
@@ -15,7 +16,12 @@ var validIdentifiers = regexp.MustCompile(`^[\p{Ll}\p{Lt}\p{Lm}\p{Lo}][\p{Ll}\p{
 var startsWithLetter = regexp.MustCompile(`^[^\p{Ll}\p{Lt}\p{Lm}\p{Lo}]`)
 var invalidCharacter = regexp.MustCompile(`([^\p{Ll}\p{Lt}\p{Lm}\p{Lo}\d\-_.])`)
 
-func ValidateIdentifier(identifier string, context ...string) *ConfigWarning {
+func ValidateIdentifier(identifier string, context ...string) (*ConfigWarning, error) {
+	var err error
+	if identifier == "" {
+		return nil, errors.New("identifier cannot be an empty string")
+	}
+
 	if identifier != "" && !validIdentifiers.MatchString(identifier) {
```

Comment:
Nit, but you can get rid of the first condition

```suggestion
	if !validIdentifiers.MatchString(identifier) {
```